### ISSUE TITLE
[Bug]: Serious issues with subgraph splitting + running graphs via rivet-node

### ISSUE THREAD

**Issue by ai-made-approachable** [Ironclad/rivet#349]
### What happened?

I have a subgraph that is succesfully running, but the output of the subgraph returns that it has 4x not run instead.

![image](https://github.com/Ironclad/rivet/assets/149181237/585a4f8f-e1e1-4424-b45b-664e0e7e4055)

Looks like if the 1st iteration is "Not ran" that all other iterations will also be "Not ran" even if that is not the case:
![image](https://github.com/Ironclad/rivet/assets/149181237/8f035703-86c8-48eb-ac11-869e4d200791)
![image](https://github.com/Ironclad/rivet/assets/149181237/cdb3c895-aad5-4e92-bec4-921e2859833e)

This issue does not happen if the 1st iteration returns something and later iterations do not return something.

### What was the expected functionality?

I like to be able to use splitting + subgraph to filter data.

### Describe your environment

MacOS, Node 14.0.0, etc.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by abrenneke** [comment]
Yeah, split and "Not Ran" don't interact well together, unfortunately. I'm not sure how to fix it correctly yet.
